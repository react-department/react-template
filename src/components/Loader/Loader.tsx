import { testIdLoader } from '../../constants/TestId';

import type { ReactElement } from 'react';

import loader from '../../assets/images/loader.svg';

import styles from './sass/Loader.module.scss';

/**
 * Primary UI component for displaying a loader in the project.
 */
export default function Loader(): ReactElement {
  return (
    <div>
      <img
        src={loader}
        className={styles.appLoader}
        alt="loader"
        data-testid={testIdLoader}
      />
    </div>
  );
}
