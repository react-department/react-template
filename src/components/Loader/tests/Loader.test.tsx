import { render } from '../../../../__tests__/test-utils';
import Loader from '../Loader';

describe('<Loader />', () => {
  test('should render component', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
