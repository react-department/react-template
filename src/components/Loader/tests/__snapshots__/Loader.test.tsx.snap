// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render component 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="loader"
      class="appLoader"
      data-testid="loader"
      src="loader.svg"
    />
  </div>
</DocumentFragment>
`;
